### PR TITLE
52 standardise optional notation

### PR DESCRIPTION
**Pull request by kartikeyakirar**
part of https://github.com/insightsengineering/nestdevs-tasks/issues/52

### CODE REVIEW

File: R/FilterStates-utils.R [hunk 1]
```diff
@@ -10,9 +10,9 @@
 #' @param dataname (`character(1)`)
 #'   name of the data used in the subset expression,
 #'   passed to the function argument attached to this `FilterStates`.
-#' @param datalabel (`character(1)`)
-#'   optional text label.
-#' @param ... (optional)
+#' @param datalabel (`character(1)`) optional,
+#'   text label.
```
Comment (m7pr) [inline R/FilterStates-utils.R:14]:
```suggestion
#' @param datalabel (`character(1)`) optional
#'   text label.
```

File: R/FilterStatesDF.R [hunk 2]
```diff
@@ -174,8 +174,8 @@ DFFilterStates <- R6::R6Class( # nolint
     #' @param dataname (`character`)
     #'   name of the data used in the *subset expression*.
     #'   Passed to the function argument attached to this `FilterStates`.
-    #' @param datalabel (`character(1)`)
-    #'   optional text label.
+    #' @param datalabel (`character(1)`) optional,
+    #'   text label.
```
Comment (m7pr) [inline R/FilterStatesDF.R:178]:
```suggestion
    #' @param datalabel (`character(1)`) optional
    #'   text label.
```

File: R/FilterStatesMAE.R [hunk 3]
```diff
@@ -24,8 +24,8 @@ MAEFilterStates <- R6::R6Class( # nolint
     #' @param dataname (`character(1)`)
     #'   name of the data used in the subset expression.
     #'   Passed to the function argument attached to this `FilterStates`.
-    #' @param datalabel (`character(1)`)
-    #'   optional text label.
+    #' @param datalabel (`character(1)`) optional,
+    #'   text label.
```
Comment (m7pr) [inline R/FilterStatesMAE.R:28]:
```suggestion
    #' @param datalabel (`character(1)`) optional
    #'   text label.
```

File: R/FilterStatesSE.R [hunk 4]
```diff
@@ -25,8 +25,8 @@ SEFilterStates <- R6::R6Class( # nolint
     #' @param dataname (`character(1)`)
     #'   name of the data used in the expression
     #'   specified to the function argument attached to this `FilterStates`.
-    #' @param datalabel (`character(1)`)
-    #'   optional text label. Should be the name of experiment.
+    #' @param datalabel (`character(1)`) optional,
+    #'   text label. Should be the name of experiment.
```
Comment (m7pr) [inline R/FilterStatesSE.R:29]:
```suggestion
    #' @param datalabel (`character(1)`) optional
    #'   text label. Should be the name of experiment.
```

File: R/FilteredData.R [hunk 5]
```diff
@@ -100,7 +100,7 @@ FilteredData <- R6::R6Class( # nolint
     #' @param data_objects (`named list`)
     #'   List of data objects.
     #'   Names of the list will be used as `dataname` for respective datasets.
-    #' @param join_keys (`join_keys`) optional joining keys, see [`teal.data::join_keys()`].
+    #' @param join_keys (`join_keys`) optional, joining keys, see [`teal.data::join_keys()`].
```
Comment (m7pr) [inline R/FilteredData.R:103]:
```suggestion
    #' @param join_keys (`join_keys`) optional joining keys, see [`teal.data::join_keys()`].
```

File: R/FilteredDataset-utils.R [hunk 6]
```diff
@@ -4,9 +4,9 @@
 #'
 #' @param dataset any object
 #' @param dataname (`character(1)`)
-#'  syntactically valid name given to the dataset.
-#' @param keys (`character`)
-#'   optional vector of primary key column names.
+#'   syntactically valid name given to the dataset.
+#' @param keys (`character`) optional,
+#'   vector of primary key column names.
```
Comment (m7pr) [inline R/FilteredDataset-utils.R:9]:
```suggestion
#' @param keys (`character`) optional
#'   vector of primary key column names.
```

File: R/FilteredDataset.R [hunk 7]
```diff
@@ -22,8 +22,8 @@ FilteredDataset <- R6::R6Class( # nolint
     #' @param dataset any object
     #' @param dataname (`character(1)`)
     #'  syntactically valid name given to the dataset.
-    #' @param keys (`character`)
-    #'   optional vector of primary key column names.
+    #' @param keys (`character`) optional,
+    #'   vector of primary key column names.
```
Comment (m7pr) [inline R/FilteredDataset.R:26]:
```suggestion
    #' @param keys (`character`) optional
    #'   vector of primary key column names.
```

File: R/FilteredDatasetDataframe.R [hunk 8]
```diff
@@ -45,8 +45,8 @@ DataframeFilteredDataset <- R6::R6Class( # nolint
     #'  single `data.frame` for which filters are rendered.
     #' @param dataname (`character(1)`)
     #'  syntactically valid name given to the dataset.
-    #' @param keys (`character`)
-    #'   optional vector of primary key column names.
+    #' @param keys (`character`) optional,
+    #'  vector of primary key column names.
```
Comment (m7pr) [inline R/FilteredDatasetDataframe.R:49]:
```suggestion
    #' @param keys (`character`) optional
    #'  vector of primary key column names.
```

File: R/FilteredDatasetMAE.R [hunk 9]
```diff
@@ -44,8 +44,8 @@ MAEFilteredDataset <- R6::R6Class( # nolint
     #'  single `MulitiAssayExperiment` for which filters are rendered.
     #' @param dataname (`character(1)`)
     #'  syntactically valid name given to the dataset.
-    #' @param keys (`character`)
-    #'   optional vector of primary key column names.
+    #' @param keys (`character`) optional,
+    #'   vector of primary key column names.
```
Comment (m7pr) [inline R/FilteredDatasetMAE.R:48]:
```suggestion
    #' @param keys (`character`) optional
    #'   vector of primary key column names.
```

File: R/teal_slice.R [hunk 10]
```diff
@@ -55,18 +55,18 @@
 #'  Must be a valid `R` expression which can be evaluated in the context of the data set.
 #'  For a `data.frame` `var == "x"` is sufficient, but `MultiAssayExperiment::subsetByColData`
 #'  requires `dataname` prefix, *e.g.* `data$var == "x"`.
-#' @param choices (optional `vector`) specifying allowed choices;
+#' @param choices (`vector`) optional, specifying allowed choices;
```
Comment (m7pr) [inline R/teal_slice.R:58]: This lost the meaning

> vector optional, specifying allowed choices;

?

We need to come up with something better in here.

File: R/teal_slice.R [hunk 11]
```diff
@@ -55,18 +55,18 @@
...
 #' When specified it should be a subset of values in variable denoted by `varname`;
 #' Type and size depends on variable type. Factors are coerced to character.
-#' @param selected (optional `vector`) of selected values from `choices`;
+#' @param selected (`vector`) optional, of selected values from `choices`;
```
Comment (m7pr) [inline R/teal_slice.R:61]: same as https://github.com/insightsengineering/teal.slice/pull/567/files#r1524711544

File: R/teal_slice.R [hunk 12]
```diff
@@ -55,18 +55,18 @@
...
 #' Type and size depends on variable type. Factors are coerced to character.
-#' @param multiple (optional `logical(1)`) flag specifying whether more than one value can be selected;
+#' @param multiple (`logical(1)`) optional, flag specifying whether more than one value can be selected;
```
Comment (m7pr) [inline R/teal_slice.R:63]:
```suggestion
#' @param multiple (`logical(1)`) optional flag specifying whether more than one value can be selected;
```

File: R/teal_slice.R [hunk 13]
```diff
@@ -55,18 +55,18 @@
...
 #' only applicable to `ChoicesFilterState` and `LogicalFilterState`
-#' @param keep_na (optional `logical(1)`) flag specifying whether to keep missing values
-#' @param keep_inf (optional `logical(1)`) flag specifying whether to keep infinite values
+#' @param keep_na (`logical(1)`) optional, flag specifying whether to keep missing values
+#' @param keep_inf (`logical(1)`) optional, flag specifying whether to keep infinite values
```
Comment (m7pr) [inline R/teal_slice.R:66]:
```suggestion
#' @param keep_na (`logical(1)`) optional flag specifying whether to keep missing values
#' @param keep_inf (`logical(1)`) optional flag specifying whether to keep infinite values
```

File: R/teal_slice.R [hunk 14]
```diff
@@ -55,18 +55,18 @@
...
 #' @param fixed (`logical(1)`) flag specifying whether to fix this filter state (forbid setting state)
 #' @param anchored (`logical(1)`) flag specifying whether to lock this filter state (forbid removing and inactivating)
-#' @param title (optional `character(1)`) title of the filter. Ignored when `varname` is set.
+#' @param title (`character(1)`) optional, title of the filter. Ignored when `varname` is set.
```
Comment (m7pr) [inline R/teal_slice.R:69]:
```suggestion
#' @param title (`character(1)`) optional title of the filter. Ignored when `varname` is set.
```

**Review by m7pr — CHANGES_REQUESTED** [review]
Hey, there are 2 places that requires a re-write. In other suggestions I proposed to remove commas before nouns.

File: R/teal_slice.R (hunk 11 above)
Comment (kartikeyakirar) [reply R/teal_slice.R]: [cb8b07e](https://github.com/insightsengineering/teal.slice/pull/567/commits/cb8b07eded85e1ab7973fc3afb2e97b4c0077237)

File: R/FilteredDataset-utils.R [hunk 15]
```diff
@@ -4,9 +4,9 @@
 #'
 #' @param dataset any object
 #' @param dataname (`character(1)`)
-#'  syntactically valid name given to the dataset.
-#' @param keys (`character`)
-#'   optional vector of primary key column names.
+#'   syntactically valid name given to the dataset.
```
Comment (m7pr) [inline R/FilteredDataset-utils.R:7]:
```suggestion
#'  syntactically valid name given to the dataset.
```

**Review by m7pr — APPROVED** [review]

**Pull request merged**